data/json/itemgroups/vending_machines.json

```
@@ -130,5 +130,303 @@
+      { "group": "vending_items_electronics", "prob": 100 }
+    ]
+  },
+  {
+    "type": "item_group",
+    "subtype": "collection",
+    "id": "vending_toys",
+    "entries": [
+      { "group": "crate_toys", "prob": 100 },
+      { "group": "crate_toys", "prob": 100 },
+      { "group": "games", "prob": 100 },
+      { "group": "games", "prob": 100 }
+    ]
+  },
+  {
+    "type": "item_group",
+    "subtype": "collection",
+    "id": "vending_maps",
+    "entries": [
+      { "group": "maps", "prob": 100 },
+      { "group": "newspaper", "prob": 100 },
+      { "group": "magazines", "prob": 100 },
+      { "group": "maps", "prob": 100 },
+      { "group": "newspaper", "prob": 100 },
+      { "group": "magazines", "prob": 100 },
+      { "group": "maps", "prob": 100 },
+      { "group": "newspaper", "prob": 100 },
+      { "group": "magazines", "prob": 100 },
+      { "group": "maps", "prob": 100 },
+      { "group": "newspaper", "prob": 100 },
+      { "group": "magazines", "prob": 100 },
+      { "group": "maps", "prob": 100 },
+      { "group": "newspaper", "prob": 100 },
+      { "group": "magazines", "prob": 100 },
+      { "group": "maps", "prob": 100 },
+      { "group": "newspaper", "prob": 100 },
+      { "group": "magazines", "prob": 100 }
+    ]
+  },
+  {
+    "type": "item_group",
+    "subtype": "collection",
+    "id": "vending_adult",
+    "entries": [
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 },
+      { "group": "vending_items_adult", "prob": 100 }
+    ]
+  },
+  {
+    "type": "item_group",
+    "subtype": "collection",
+    "id": "vending_bionics",
+    "entries": [
+      { "group": "bionics_common", "prob": 100 },
+      { "group": "bionics_common", "prob": 100 },
+      { "group": "bionics_common", "prob": 100 },
+      { "group": "bionics_common", "prob": 100 },
+      { "group": "bionics_common", "prob": 100 },
+      { "group": "tools_robotics", "prob": 50 },
+      { "group": "bionics", "prob": 20 }
```

Robotics tools contains some things like hauler bots, teleporters, control laptops etc that all should not be in a vending machine in public, and likewise the generic `bionics` group contains all bionics, including/especially military/lab-only ones. Meanwhile, this spawns no power storage CBMs, so I'd suggest replacing them with some potential spawns of that:
```suggestion
      { "group": "bionic_power_storage_civ", "prob": 100 },
      { "group": "bionic_power_storage_civ", "prob": 80 },
      { "group": "bionic_power_storage_civ", "prob": 80 },
      { "group": "bionic_power_storage_civ", "prob": 60 },
      { "group": "bionic_power_storage_civ", "prob": 60 },
```

---

yeah this was a big one i definitely knew would need changed, sending it through